src/LoadingSparseTaskS3.cpp

```diff
@@ -26,16 +26,11 @@ SparseDataset LoadingSparseTaskS3::read_dataset(
   }
 
   // READ the kaggle criteo dataset
-  return input.read_input_criteo_kaggle_sparse(
-      config.get_input_path(),
-      delimiter,
-      config);
+  return input.read_input_criteo_kaggle_sparse(config.get_input_path(),
+                                               delimiter, config);
 }
 
 void LoadingSparseTaskS3::check_label(FEATURE_TYPE label) {
-  if (label != 1.0 && label != 0.0) {
```

Comment:
I think we should check differently depending on the type of dataset. Removing these checks doesn't seems a good idea.